.circleci/verbatim-sources/linux-build-defaults.yml

```diff
@@ -42,13 +42,15 @@ pytorch_linux_build_defaults: &pytorch_linux_build_defaults
 
         # Push intermediate Docker image for next phase to use
         if [ -z "${BUILD_ONLY}" ]; then
-          # Note [namedtensor build image]
-          # The namedtensor build uses the same docker image as
+          # Note [Special build images]
+          # The namedtensor and xla builds use the same docker image as
           # pytorch-linux-trusty-py3.6-gcc5.4-build. In the push step, we have to
-          # distinguish between these two so the test can pick up the correct image.
+          # distinguish between them so the test can pick up the correct image.
           output_image=${DOCKER_IMAGE}-${CIRCLE_SHA1}
           if [[ ${BUILD_ENVIRONMENT} == *"namedtensor"* ]]; then
             export COMMIT_DOCKER_IMAGE=$output_image-namedtensor
+          elif [[ ${BUILD_ENVIRONMENT} == *"xla"* ]]; then
+            export COMMIT_DOCKER_IMAGE=$output_image-xla
```

Comment:
There are two more robust long-term solutions; the current one in this PR is a bit of a hack. They are:
1) don't special case the XLA / named tensor builds. The named tensor build, at least, is a temporary build that I'm planning on removing sometime down the line, but maybe the XLA one should get its own config and use a different docker image?
2) change the output image to the name of the build.